### PR TITLE
chore: remove todo storage client grpc flag

### PR DESCRIPTION
**Pull request by bgdnvk**
You already pushed this into main so this TODO shouldn't be here.

### CODE REVIEW

File: storage/storage.go [hunk 1]
```diff
@@ -119,7 +119,6 @@ type Client struct {
 	tc storageClient
 	// useGRPC flags whether the client uses gRPC. This is needed while the
 	// integration piece is only partially complete.
-	// TODO: remove before merging to main.
```
Comment (tritone) [inline storage/storage.go:122]: Could you also remove the field (`useGRPC`) which it is referencing?

Looks like the only reference is on L240 of this file.

**Review by tritone — COMMENTED** [review]
Thanks!

**Review by tritone — APPROVED** [review]

**Review by tritone — APPROVED** [review]

**Comment by bgdnvk** [comment]
From the logs:
```
    --- FAIL: TestRetryConformance/http-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1 (0.01s)
    --- PASS: TestRetryConformance/grpc-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1 (0.57s)
```

looks like an http insert has failed

**Comment by tritone** [comment]
> From the logs:
> 
> ```
>     --- FAIL: TestRetryConformance/http-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1 (0.01s)
>     --- PASS: TestRetryConformance/grpc-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1 (0.57s)
> ```
> 
> looks like an http insert has failed

This is most likely a test flake; I'll rerun.

**Comment by bgdnvk** [comment]
Failed again, might be an edge case since all others pass

`    --- FAIL: TestRetryConformance/http-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1 (0.03s)
`
from what i understand it's just an http connection and the client options haven't changed, the flag wasn't used there - perhaps in another test somehow? can dig more into it but im unfamiliar with the codebase

**Comment by bgdnvk** [comment]
The one failing is from the "earliest version"
```
[15:09:59 PDT] Child job: cloud-devrel/client-libraries/go/google-cloud-go/presubmit/latest-version finished with status: SUCCEEDED Url: https://source.cloud.google.com/results/invocations/e91b0f84-8d26-4ccd-a4c9-977173d74780
[15:10:34 PDT] Child job: cloud-devrel/client-libraries/go/google-cloud-go/presubmit/earliest-version finished with status: FAILED Url: https://source.cloud.google.com/results/invocations/9b23c3c7-4126-432c-8aab-6d45e97fbc5c
```

however from the successful run we have
```
=== RUN   TestRetryConformance/http-2-[return-reset-connection_return-reset-connection]-storage.objects.insert-1
    retry_conformance_test.go:634: want success, got Writer.Close: Post "http://localhost:9000/upload/storage/v1/b/bucket-20240517-79706071112773-0116/o?alt=json&ifGenerationMatch=0&name=new-object.txt&prettyPrint=false&projection=full&uploadType=multipart": write tcp 127.0.0.1:58030->127.0.0.1:9000: write: broken pipe
```
